lib/application.js

```diff
@@ -158,6 +158,11 @@ module.exports = class Application extends Emitter {
       keys: this.keys,
       secure: request.secure
     });
+    if (req && req.headers) {
+      request.ip = request.ips[0] || req.socket.remoteAddress || '';
+    } else {
+      request.ip = ''; // We are in a badly written test :P
```

Comment:
Well I think the nice thing would be to guarantee that every request has an ip. That removes the burden from users of the library to wonder (and check) if it's  ''. I don't really see any reason why a request should be processed that can't get an ip